packages/core/src/controllers/events.ts

```diff
@@ -0,0 +1,200 @@
+import { generateChildLogger, Logger } from "@walletconnect/logger";
+import { ICore, IEventClient, EventClientTypes } from "@walletconnect/types";
+import { uuidv4 } from "@walletconnect/utils";
+import {
+  CORE_STORAGE_PREFIX,
+  EVENTS_CLIENT_API_URL,
+  EVENTS_STORAGE_CLEANUP_INTERVAL,
+  EVENTS_STORAGE_CONTEXT,
+  EVENTS_STORAGE_VERSION,
+  RELAYER_SDK_VERSION,
+} from "../constants";
+import { HEARTBEAT_EVENTS } from "@walletconnect/heartbeat";
+import { fromMiliseconds } from "@walletconnect/time";
+
+export class EventClient extends IEventClient {
+  public readonly context = EVENTS_STORAGE_CONTEXT;
+  private readonly storagePrefix = CORE_STORAGE_PREFIX;
+  private readonly storageVersion = EVENTS_STORAGE_VERSION;
+
+  private events = new Map<string, EventClientTypes.Event>();
+  private toPersist = false;
+  constructor(public core: ICore, public logger: Logger, telemetryEnabled = true) {
+    super(core, logger, telemetryEnabled);
+    this.logger = generateChildLogger(logger, this.context);
+    if (telemetryEnabled) {
+      this.restore().then(async () => {
+        await this.submit();
+        this.setEventListeners();
+      });
+    } else {
+      // overwrite any persisted events with an empty array
+      this.persist();
+    }
+  }
+
+  get storageKey() {
+    return (
+      this.storagePrefix + this.storageVersion + this.core.customStoragePrefix + "//" + this.context
+    );
+  }
+
+  public createEvent: IEventClient["createEvent"] = (params) => {
+    const {
+      event = "ERROR",
+      type = "",
+      properties: { topic, trace },
+    } = params;
+    const eventId = uuidv4();
+    const bundleId = this.core.projectId || "";
```

Comment:
`bundleId` is for mobile platfroms, in your case you should send the `domain` of the web app using the client.

Reply:
I will leave it as is because we also cater to node environments where domain is not available